rkt/config/directories.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+type configurableDirectoriesV1 struct {
+	Data string `json:"data"`
+}
+
+func init() {
+	addParser("directories", "v1", &configurableDirectoriesV1{})
+	// Look in 'directories' subdir for configs of type directories
+	registerSubDir("dirs.d", []string{"directories"})
+}
+
+func (p *configurableDirectoriesV1) parse(config *Config, raw []byte) error {
+	var dirs configurableDirectoriesV1
+	if err := json.Unmarshal(raw, &dirs); err != nil {
+		return err
+	}
+	if config.Directories.DataDir != "" {
+		// A clash has occurred. Data dir has been defined more than once in
+		// the same directory
+		return errors.New("Error processing the data directory. Verify that it has not been defined more than once in the same config directory.\n")
```

Comment:
We usually use `fmt.Errorf()` for error reporting. Also, please remove the newline - the code that actually prints the error should worry about newlines.